src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -39,59 +33,94 @@ const propTypes = {
     A positive value shifts the tooltip down, and a negative value shifts it up. */
     shiftVertical: PropTypes.number.isRequired,
 
-    /** Callback to set the Ref to the Tooltip */
-    setTooltipRef: PropTypes.func.isRequired,
-
     /** Text to be shown in the tooltip */
     text: PropTypes.string.isRequired,
 
-    /** Callback to be used to calulate the width and height of tooltip */
-    measureTooltip: PropTypes.func.isRequired,
-};
+    /** Number of pixels to set max-width on tooltip  */
+    maxWidth: PropTypes.number.isRequired,
 
-const defaultProps = {};
-
-const TooltipRenderedOnPageBody = (props) => {
-    const {
-        animationStyle,
-        tooltipWrapperStyle,
-        tooltipTextStyle,
-        pointerWrapperStyle,
-        pointerStyle,
-    } = getTooltipStyles(
-        props.animation,
-        props.windowWidth,
-        props.xOffset,
-        props.yOffset,
-        props.wrapperWidth,
-        props.wrapperHeight,
-        props.tooltipWidth,
-        props.tooltipHeight,
-        props.shiftHorizontal,
-        props.shiftVertical,
-    );
-    return ReactDOM.createPortal(
-        <Animated.View
-            ref={props.setTooltipRef}
-            onLayout={props.measureTooltip}
-            style={[tooltipWrapperStyle, animationStyle]}
-        >
-            <Text style={tooltipTextStyle} numberOfLines={1}>{props.text}</Text>
-            <View style={pointerWrapperStyle}>
-                <View style={pointerStyle} />
-            </View>
-        </Animated.View>,
-        document.querySelector('body'),
-    );
+    /** Maximum number of lines to set on tooltip */
+    numberOfLines: PropTypes.number.isRequired,
 };
 
-TooltipRenderedOnPageBody.propTypes = propTypes;
-TooltipRenderedOnPageBody.defaultProps = defaultProps;
-TooltipRenderedOnPageBody.displayName = 'TooltipRenderedOnPageBody';
-
 // Props will change frequently.
 // On every tooltip hover, we update the position in state which will result in re-rendering.
 // We also update the state on layout changes which will be triggered often.
 // There will be n number of tooltip components in the page.
 // Its good to memorize this one.
```

Comment:
I know this isn't your comment, but fixing this is helpful while we're here :D

```suggestion
// It's good to memorize this one.
```

Reply:
Done.